src/openstudio_lib/ThermalZonesGridView.cpp

```diff
@@ -470,42 +464,40 @@ void ThermalZonesGridController::addColumns(const QString& /*category*/, std::ve
 
       std::function<boost::optional<model::Schedule>(model::ThermalZone*)> humidifyingSchedule([](model::ThermalZone* z) {
         boost::optional<model::Schedule> schedule;
-        if (boost::optional<model::ZoneControlHumidistat> thermostat = z->zoneControlHumidistat()) {
-          schedule = thermostat->humidifyingRelativeHumiditySetpointSchedule();
+        if (boost::optional<model::ZoneControlHumidistat> humidistat = z->zoneControlHumidistat()) {
+          schedule = humidistat->humidifyingRelativeHumiditySetpointSchedule();
         }
         return schedule;
       });
 
       std::function<bool(model::ThermalZone*, const model::Schedule&)> setHumidifyingSchedule([](model::ThermalZone* z, model::Schedule t_s) {
-        if (boost::optional<model::ZoneControlHumidistat> thermostat = z->zoneControlHumidistat()) {
-          return thermostat->setHumidifyingRelativeHumiditySetpointSchedule(t_s);
+        bool result = false;
+        if (boost::optional<model::ZoneControlHumidistat> humidistat = z->zoneControlHumidistat()) {
+          result = humidistat->setHumidifyingRelativeHumiditySetpointSchedule(t_s);
         } else {
-          model::ZoneControlHumidistat t(z->model());
-          z->setZoneControlHumidistat(t);
-          return t.setHumidifyingRelativeHumiditySetpointSchedule(t_s);
+          model::ZoneControlHumidistat h(z->model());
+          result = h.setHumidifyingRelativeHumiditySetpointSchedule(t_s);
+          z->setZoneControlHumidistat(h);
         }
+        return result;
       });
 
       boost::optional<std::function<void(model::ThermalZone*)>> resetHumidifyingSchedule([](model::ThermalZone* z) {
-        if (boost::optional<model::ZoneControlHumidistat> thermostat = z->zoneControlHumidistat()) {
-          thermostat->resetHumidifyingRelativeHumiditySetpointSchedule();
-          if (!thermostat->dehumidifyingRelativeHumiditySetpointSchedule()) {
-            thermostat->remove();
+        if (boost::optional<model::ZoneControlHumidistat> humidistat = z->zoneControlHumidistat()) {
+          humidistat->resetHumidifyingRelativeHumiditySetpointSchedule();
+          if (!humidistat->dehumidifyingRelativeHumiditySetpointSchedule()) {
+            z->resetZoneControlHumidistat();
```

Comment:
I had a doubt about this. But this does remove it too, so no problem: https://github.com/NREL/OpenStudio/blob/db1d522abee3d173e904f5c78a716c6274dfb092/src/model/ThermalZone.cpp#L1179-L1185

Reply:
Yeah, I checked that too.  Just removing the humidistat does not trigger change signals from the zone object.  I guess now that we are properly watching the humidistat all the time we could go back to the old way because we would refresh when the humidistat was removed.  Let me know if you want me to change that?  It would protect us from future breakages if OpenStudio ever changes `resetZoneControlHumidistat` to not remove the thermostat.